superbench/benchmarks/micro_benchmarks/disk_performance.py

```diff
@@ -0,0 +1,222 @@
+                    '--%s_numjobs' % io_str,
+                    type=int,
+                    default=default_numjobs,
+                    required=False,
+                    help='Number of threads in %s test.' % io_str,
+                )
+
+    def _preprocess(self):
+        """Preprocess/preparation operations before the benchmarking.
+
+        Return:
+            True if _preprocess() succeed.
+        """
+        if not super()._preprocess():
+            return False
+
+        fio_path = os.path.join(self._args.bin_dir, self._bin_name)
+
+        for block_device in self._args.block_devices:
+            if not Path(block_device).is_block_device():
+                self._result.set_return_code(ReturnCode.INVALID_ARGUMENT)
+                logger.error('Invalid block device: {}.'.format(block_device))
+                return False
+
+            if self._args.enable_seq_precond:
+                command = fio_path +\
+                    ' --filename=%s' % block_device +\
+                    self._fio_args['seq_precond']
+                self._commands.append(command)
+
+            if self._args.rand_precond_time > 0:
+                command = fio_path +\
+                    ' --filename=%s' % block_device +\
+                    ' --runtime=%ds' % self._args.rand_precond_time +\
+                    self._fio_args['rand_precond']
+                self._commands.append(command)
+
+            for io_pattern in self._io_patterns:
+                for io_type in self._io_types:
+                    io_str = '%s_%s' % (io_pattern, io_type)
+                    runtime = getattr(self._args, '%s_runtime' % io_str)
+                    if runtime > 0:
+                        command = fio_path +\
+                            ' --filename=%s' % block_device +\
+                            ' --ramp_time=%ds' % getattr(self._args, '%s_ramp_time' % io_str) +\
+                            ' --runtime=%ds' % runtime +\
+                            ' --iodepth=%d' % getattr(self._args, '%s_iodepth' % io_str) +\
+                            ' --numjobs=%d' % getattr(self._args, '%s_numjobs' % io_str) +\
+                            self._fio_args[io_str]
+                        self._commands.append(command)
+
+        return True
+
+    def _process_raw_result(self, cmd_idx, raw_output):
+        """Function to parse raw results and save the summarized results.
+
+          self._result.add_raw_data() and self._result.add_result() need to be called to save the results.
+
+        Args:
+            cmd_idx (int): the index of command corresponding with the raw_output.
+            raw_output (str): raw output string of the micro-benchmark.
+
+        Return:
+            True if the raw output string is valid and result can be extracted.
+        """
+        self._result.add_raw_data('raw_output_' + str(cmd_idx), raw_output)
+
+        try:
+            fio_output = json.loads(raw_output)
+        except BaseException as e:
```

Comment:
How about letting 'try' cover more code? If the output format changed and some field are missing, the following code may raise 'KeyError'

Reply:
Fixed.